lib/neovim-plugin.nix

```diff
@@ -99,21 +102,36 @@ with lib;
             example = settingsExample;
           };
         }
+        // optionalAttrs allowLazyLoad {
+          lazyLoad = nixvimOptions.mkLazyLoadOption {
+            inherit originalName cfg;
+            optionsForPlugin = true;
+            lazyLoad = {
+              # TODO: extract extraConfigLua and extraConfigLuaPre from extraConfig
+              after = optionalString callSetup ''
+                require('${luaName}')${setup}(${optionalString (cfg ? settings) (toLuaObject cfg.settings)})
+              '';
+              colorscheme = if (isColorscheme && (colorscheme != null)) then [ colorscheme ] else null;
```

Comment:
Could the `colorscheme` arg be merged in with `optionalAttrs` rather than explicitly passing in `null`

```nix
// (optionalAttrs (isColorscheme && colorscheme != null) { colorscheme = [ colorscheme ]; })
```

Also, why does mkLazyLoadOption take colorscheme as a list? If it accepted string values (or either using `lib.toList`) then we could do `{ inherit colorscheme; }`.

Reply:
`optionalAttrs` 

> Could the `colorscheme` arg be merged in with `optionalAttrs` rather than explicitly passing in `null`
> 
> ```nix
> // (optionalAttrs (isColorscheme && colorscheme != null) { colorscheme = [ colorscheme ]; })
> ```
Yes.

> Also, why does mkLazyLoadOption take colorscheme as a list? If it accepted string values (or either using `lib.toList`) then we could do `{ inherit colorscheme; }`.
The colorscheme option is a list because of merging. 
It could be of type `(either str (listOf str))`. But if I understand correctly, when it is defined as a `str`, then it cannot be merged with other values set as `str` or `list`.
In that case, if plugin mantainers set the option as a `str`, then user cannot extend it.

Reply:
> The colorscheme option is a list because of merging.

Ah, I read this as passing in arguments to mkLazyLoadOption rather than defining config.

It's still possible to do that with the type system, though whether we need to in this case is another matter!

Various non-list types can be merged (e.g. types.lines), however the simplist option here would be a type that coerces to a list:

```nix
type =
  with lib.types;
  coercedTo str lib.toString (listOf str);
```

Reply:
I didn't know about `coercedTo`. I am going to implement it that way, and also the other `str or listOf str`. Thanks for the help